client/my-sites/patterns/components/pattern-library/index.tsx

```diff
@@ -159,6 +160,29 @@ export const PatternLibrary = ( {
 		setSearchFormKey( category );
 	}, [ category ] );
 
+	const [ isSticky, setIsSticky ] = useState( false );
+	const navRef = useRef< HTMLElement >( null );
+	const prevNavTopValue = useRef( 0 );
+
+	const handleScroll = () => {
+		if ( ! navRef.current ) {
+			return;
+		}
+
+		const navbarPosition = navRef.current.getBoundingClientRect().top;
+
+		setIsSticky( navbarPosition === prevNavTopValue.current );
+
+		prevNavTopValue.current = navbarPosition;
+	};
+
+	useEffect( () => {
+		window.addEventListener( 'scroll', handleScroll );
+		return () => {
+			window.removeEventListener( 'scroll', handleScroll );
+		};
+	}, [] );
```

Comment:
```suggestion
	useEffect( () => {
		const handleScroll = () => {
			if ( ! navRef.current ) {
				return;
			}

			const navbarPosition = navRef.current.getBoundingClientRect().top;

			setIsSticky( navbarPosition === prevNavTopValue.current );

			prevNavTopValue.current = navbarPosition;
		};

		window.addEventListener( 'scroll', handleScroll, { passive: true } );

		return () => {
			window.removeEventListener( 'scroll', handleScroll );
		};
	}, [] );
```
`handleScroll` should be defined within the `useEffect` callback. Or is there some reason not to do this?

We should also pass `{ passive: true }` to the `addEventListener` call for good measure.